src/utils.jl

```diff
@@ -44,8 +44,13 @@ users(d::AbstractDict) = field(d, "users")
 market(d::AbstractDict) = field(d, "market")
 "Function to get the profile dictionary"
 profiles(d::AbstractDict) = field_d(d, "profile")
+"Auxiliary function to check if the key 'type' is available in the dictionary d, otherwise false"
+has_type(d::AbstractDict) = ("type" in keys(d))
+has_type(d) = false  # if d is not an abstract dictionary, then return false
 "Function to get the components list of a dictionary"
-components(d::AbstractDict) = d
+function components(d::AbstractDict)
+      return Dict(k=>v for (k,v) in d if has_type(v))

```

Comment:
Did you add a space or two tabs by mistake? return doesn't seem aligned with the other tabs